File: app/src/main/java/io/euphoria/xkcd/app/ui/RoomUIManager.java
package io.euphoria.xkcd.app.ui;

/** Created by Xyzzy on 2017-02-24. */

/* Room UI wrapper retrieval hub */
public interface RoomUIManager {

    /* Set the factory for creating RoomUI-s */
    void setRoomUIFactory(RoomUIFactory factory);

    /* Get the room UI for the given name */
    RoomUI getRoomUI(String roomName);

    /* Install an event listener */
    void addEventListener(UIManagerListener l);

    /* Remove an event listener */
    void removeEventListener(UIManagerListener l);

}

File: app/src/main/java/io/euphoria/xkcd/app/ui/RoomUIManager.kt
package io.euphoria.xkcd.app.ui

/** Created by Xyzzy on 2017-02-24.  */ /* Room UI wrapper retrieval hub */
interface RoomUIManager {
    /* Set the factory for creating RoomUI-s */
    fun setRoomUIFactory(factory: RoomUIFactory?)

    /* Get the room UI for the given name */
    fun getRoomUI(roomName: String): RoomUI

    /* Install an event listener */
    fun addEventListener(l: UIManagerListener)

    /* Remove an event listener */
    fun removeEventListener(l: UIManagerListener)
}
